### DISCUSSION TITLE
failed to create task for container error on docker install

### DISCUSSION THREAD

**Discussion by ATLAS2246** [inventree/InvenTree#7401] [Q&A] [answered]
Im attempting to install a production version of Inventree on a rasberry pi 4 running rasberry pi os. ive followed the docker install guide and when i run docker compose up -d i get 

WARN[0000] /home/QPuter/Inventree/docker-compose.yml: `version` is obsolete 
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.45/containers/json?all=1&filters=%7B%22label%22%3A%7B%22com.docker.compose.config-hash%22%3Atrue%2C%22com.docker.compose.project%3Dinventree%22%3Atrue%7D%7D": dial unix /var/run/docker.sock: connect: permission denied

ive attempted to resolve this using sudo docker compose up -d but i get a different error back

WARN[0000] /home/QPuter/Inventree/docker-compose.yml: `version` is obsolete 
[+] Running 5/6
 ✔ Network inventree_default   Created                                                            0.1s 
 ✔ Container inventree-db      Started                                                            1.4s 
 ✔ Container inventree-cache   Started                                                            1.6s 
 ✔ Container inventree-server  Started                                                            1.6s 
 ✔ Container inventree-worker  Started                                                            2.3s 
 ⠼ Container inventree-proxy   Starting                                                           2.7s 
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/home/QPuter/Inventree/Caddyfile" to rootfs at "/etc/caddy/Caddyfile": mount /home/QPuter/Inventree/Caddyfile:/etc/caddy/Caddyfile (via /proc/self/fd/6), flags: 0x5000: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

ive tried to search around but i havent been able to find much help on this topic, in all fairness i am very new to this, so any help would be greatly appreciated

**Comment by SchrodingersGat** [comment]
What version of `docker` and `docker compose` are you running?

**Reply by ATLAS2246** [reply]
looking at the docker install doc, it told me to run a couple commands to find out, so ive just included everything it spat out

QPuter@raspberrypi:~ $ docker compose version
Docker Compose version v2.27.0
QPuter@raspberrypi:~ $ docker --version
Docker version 26.1.3, build b72abbb
QPuter@raspberrypi:~ $ docker version
Client: Docker Engine - Community
 Version:           26.1.3
 API version:       1.45
 Go version:        go1.21.10
 Git commit:        b72abbb
 Built:             Thu May 16 08:33:25 2024
 OS/Arch:           linux/arm64
 Context:           default
permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.45/version": dial unix /var/run/docker.sock: connect: permission denied

**Reply by ATLAS2246** [reply]
I found a solution, thanks for your help

**Comment by ATLAS2246** [comment, marked as answer]
I found a solution, when i downloaded caddyfile, it came with the extension .txt, this meant when i first ran inventree it created a directory called Caddyfile this discussion suggested this was the problem https://caddy.community/t/error-for-caddy-caddy-1-cannot-start-service-caddy-failed-to-create-shim-task/16753/3 so i deleted the directory and it booted successfully from there.

Thanks!!!

**Reply by breadx333** [reply]
This helped me. Delete the directory and remove file format .txt
